includes/core/classes/class-settings.php

```diff
@@ -691,4 +693,72 @@ public function datetime_preview( string $name, string $value ): void {
 			);
 		}
 	}
+
+	/**
+	 * Display a preview of the rewritten URL based on the specified string.
+	 *
+	 * This method is used to display a preview of the rewritten URL based on the specified
+	 * string.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @param string $name  The name of the urlrewrite format option.
+	 * @param string $value The value of the urlrewrite format option.
+	 * @return void
+	 */
+	public function urlrewrite_preview( string $name, string $value ): void {
+		if (
+			'gatherpress_general[urls][events]' === $name ||
+			'gatherpress_general[urls][venues]' === $name ||
+			'gatherpress_general[urls][topics]' === $name
+		) {
+			switch ( $name ) {
+				case 'gatherpress_general[urls][events]':
+					$suffix = _x( 'sample-event', 'sample event post slug', 'gatherpress' );
+					break;
+
+				case 'gatherpress_general[urls][venues]':
+					$suffix = _x( 'sample-venue', 'sample venue post slug', 'gatherpress' );
+					break;
+
+				case 'gatherpress_general[urls][topics]':
+					$suffix = _x( 'sample-topic-term', 'sample topic term slug', 'gatherpress' );
+					break;
+
+				default:
+					break;
+			}
+			Utility::render_template(
+				sprintf( '%s/includes/templates/admin/settings/partials/urlrewrite-preview.php', GATHERPRESS_CORE_PATH ),
+				array(
+					'name'   => $name,
+					'value'  => $value,
+					'suffix' => $suffix,
+				),
+				true
+			);
+		}
+	}
+
+	/**
+	 * Update Rewrite rules, when post type rewrite slugs change.
+	 *
+	 * Fires after the value of the 'gatherpress_general["urls"]' option-part has been successfully updated
+	 * and only if it has changed since before.
+	 *
+	 * @since 0.31.0
```

Comment:
`1.0.0` unless you want to have a conversation about this and we review all the `1.0.0` and change them to make a new baseline. For me it seems easily to just set `1.0.0` as the first version of the stable plugin.